api/v1alpha1/managedocs_types.go

```diff
@@ -38,9 +38,27 @@ type ManagedOCSSpec struct {
 	ReconcileStrategy ReconcileStrategy `json:"reconcileStrategy,omitempty"`
 }
 
+type ComponentStatus string
+
+const (
+	ComponentReady   ComponentStatus = "Ready"
+	ComponentPending ComponentStatus = "Pending"
+	ComponentNone    ComponentStatus = "None"
+)
+
+type ManagedOCSComponent struct {
+	Status      ComponentStatus `json:"status"`
+	LastChecked metav1.Time     `json:"lastChecked"`
+}
+
+type ManagedOCSComponents struct {
+	StorageCluster ManagedOCSComponent `json:"storageCluster"`
+}
```

Comment:
I will suggest renaming these structs:
-  The `ManagedOCS` prefix is unneeded as all these structures live under the ManagedOCS API
- `Components` and `Components` are easily mixed while reading (happened to me) 
- The structures' name does not covey the fact these are status structures.

I will suggest something in the line of `ComponentStatusMap` and `ComponentStatus`